src/lib.rs

```diff
@@ -46,15 +31,16 @@ fn stream_bytes(
                 .body_string("".into()));
         } else {
             let index = Path::new(actual_path).join("index.html");
-            return stream_bytes(root, &*index.to_string_lossy(), headers);
+            return stream_bytes(root, &*index.to_string_lossy()).await;
```

Comment:
I wonder if we could avoid the recursion by redirecting like in the branch above? It's probably faster if the server modifies the path itself instead of asking the connected client to make another request though. Hmmm

Reply:
Alright, I'm currently working on this, but not by redirecting. Basically I'm adding a `loop` around a large part of the body of `stream_bytes`. I'll submit a PR to your forked repo when it's ready